### ISSUE TITLE
The 'profile:username' meta tag not always injected on profile pages

### ISSUE THREAD

**Issue by bramus** [mastodon/mastodon#21591]
### Steps to reproduce the problem

1. Visit https://mastodon.social/public/local
2. Click on any profile
3. Using DevTools, run `document.querySelector('meta[property="profile:username"]')`

### Expected behaviour

For the element to be present

### Actual behaviour

The element is not present, and `null` is returned

### Detailed description

When refreshing the profile page, forcing a full reload, the meta tag is present.

### Specifications

Mastodon 4.0.2

**Comment by afontenot** [comment]
Is this surprising behavior? What are the negative consequences?

**Comment by ineffyble** [comment]
Yes, I'd like to check too. I'd expect most things that look at that meta tag to never use client side navigation.

**Comment by bramus** [comment]
> Is this surprising behavior?

Yes. The page state in the head does not reflect the actual state/what is shown. There should be no difference between SPA or MPA navigations.

> What are the negative consequences?

This leads to issues for extensions authors – [like me](https://github.com/bramus/mastodon-profile-redirect/) – who would like to read the profile username from the page. With an incorrect value they can not rely on the meta tag as it’s unpredictable.

Without this meta tag (with a correct value), extension authors would need to rely on either DOM Sniffing or URL parsing to extract profile usernames. This is not feasible as:

- DOM sniffing:
  - This breaks on instances that use a different theme _(with different markup)_ 
  - This is not future proof, as Mastodon updates might alter the markup, upon which they’ll break.
- URL parsing:
  - This does not play nice with instances that are configured with a `WEB_DOMAIN` different than the `LOCAL_DOMAIN
  - This is not future proof in case URLs ever were to change.

Please have the `'meta[property="profile:username"]'` reflect the correct value at all times, independent of what navigation type that was used, so that it’s reliable.

**Comment by joshka** [comment]
+1 I've also seen this happen when playing around with https://github.com/joshka/follow-mastodon-user-webextension/ wherein I'd like to show a page action only when a page has a username, but navigating to another page leaves the first page's meta username.
